platform/util/platform-util-uuidgenerator/util-uuidgenerator-impl/src/main/java/org/codice/ddf/platform/util/uuidgenerator/impl/UuidGeneratorImpl.java

```diff
@@ -0,0 +1,57 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.platform.util.uuidgenerator.impl;
+
+import java.util.UUID;
+import java.util.regex.Pattern;
+
+import org.apache.commons.lang3.StringUtils;
+import org.codice.ddf.platform.util.uuidgenerator.UuidGenerator;
+
+public class UuidGeneratorImpl implements UuidGenerator {
+
+    private Pattern hexPattern = Pattern.compile("^[0-9A-Fa-f]+$");
+
+    private Pattern hexPatternWithHypens = Pattern.compile("^[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}$");
+
+    private Boolean removeHyphens = true;
+
+    public void setRemoveHyphens(Boolean removeHyphens) {
+        this.removeHyphens = removeHyphens;
+    }
+
+    @Override
+    public String generateUuid() {
+        String uuid = UUID.randomUUID()
+                .toString();
+        if (removeHyphens) {
+            uuid = uuid.replaceAll("-", "");
+        }
+        return uuid;
+    }
+
+    @Override
+    public boolean validateUuid(String uuid) {
+        if (StringUtils.isEmpty(uuid)) {
+            return false;
+        }
+
+        if (uuid.length() == 32) {
```

Comment:
I can just see a scenario where we either have a discrepancy in the setting being changed or something to do with other systems interacting with us. 

It also seems odd to provide the admins an option that they can only set once (that must be before any metacards are ever created) and there is no way to go back without something custom to rewrite solr ids to be a different format. Also this prevents us from making any automatically generated metacards at system startup. 

As little as I like adding more configuration options... maybe this is a case to add a "strict validation" option thats on by default so it validates as @millerw8 is saying, but an option to turn it off and make both valid?